### PR TITLE
modules/opkg+restartcheck: test nisysapi conf.d dir for changes

### PR DESCRIPTION
**Pull request by chotea**
Upstream 3 commits in opgk/restartcheck modules from ni salt branch + added tests.

----------------------------------

 modules/restartcheck: return True if NILRT file state doesn't exist

This fixes a bug on NILinuxRT where first-time installation of sysapi
files did not trigger a reboot.

If _file_changed_nilrt is called on a file without timestamp/checksum
info it means the file was newly created/first-time installed and a
reboot is required even though the timestamp/checksum info is missing.

The timestamp/checksum info will be generated for the new file by
_update_nilrt_restart_state() in the opkg module as a result of
returning True here.

Signed-off-by: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>

------------------------------------------

 modules/opkg/restartcheck: test nisysapi conf.d dir for changes

NISysAPI expert .ini conf files can also be installed separately in a
conf.d directory as opposed to all expert configuration living in the
global .ini config file.

This change makes restartcheck and opkg aware of changes to the sysapi
conf.d by tracking the total number of files and each file's timestamp
and checksum.

Signed-off-by: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>

----------------------------------------

 modules/restartcheck: propagate previous NILRT restartcheck=True

If a reboot required event was previously witnessed on NILinuxRT as a
result of running restartcheck, propagate that restartcheck = True
result forward in subsequent restartcheck.restartcheck runs.

In other words, if once restartcheck returns True on NILinuxRT, it
will always return True until the next reboot.

This enables other software on the system which already uses salt to
check for reboots via the salt interface by calling restartcheck
directly instead searching the OS filesystem for a 'reboot_witnessed'
file.

Signed-off-by: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>

### CODE REVIEW

**Comment by chotea** [comment]
It seems we already have an OPEN PR for merging these into master: https://github.com/saltstack/salt/pull/51218
